system/lib/libc/musl/src/fcntl/fcntl.c

```diff
@@ -8,10 +8,14 @@
 int fcntl(int fd, int cmd, ...)
 {
 	unsigned long arg;
-	va_list ap;
-	va_start(ap, cmd);
-	arg = va_arg(ap, unsigned long);
-	va_end(ap);
+	// XXX Emscripten: According to the va_arg man page it is undefined behaviour to
+	// read arguments that are not passed.
+	if (cmd != F_GETFL && cmd != F_GETFD && cmd != F_GETOWN) {
```

Comment:
`F_GETOWN` does seem to read `arg`?

Reply:
(in fact, they almost all seem to - am I misreading it?)

Reply:
According the man page they don't:

```
       F_GETFD (void)
              Return (as the function result) the file descriptor flags; arg is ignored.

       F_GETFL (void)
              Return (as the function result) the file access mode and the file status flags; arg
              is ignored.

       F_GETOWN (void)
              Return  (as  the  function result) the process ID or process group ID currently re‐
              ceiving SIGIO and SIGURG signals for events on file descriptor fd.  Process IDs are
              returned as positive values; process group IDs are returned as negative values (but
              see BUGS below).  arg is ignored.
```

Notice how each of them says `arg is ignored`... and indeed in our test case we don't pass and arg to any of them.

Note that we still pass the `arg` variable to the JS side although its uninitialized and unused and never read by the JS side (it would be undefined behaviour for the JS side to read it I think).

I could make a bigger patch that changes the `syscall(SYS_fcntl` callsite too to avoid passing `arg` but that seems needless intrusive.

Reply:
that's the confusing part for me too. 
In my man it is called liked that :
```
               int flags;

               flags = fcntl(fd, F_GETFD);
               if (flags == −1)
                   /* Handle error */;
               flags |= FD_CLOEXEC;
               if (fcntl(fd, F_SETFD, flags) == −1)
                   /* Handle error */;"

```

on the GNU documentation :

```
int
set_cloexec_flag (int desc, int value)
{
  int oldflags = fcntl (desc, F_GETFD, 0);
  /* If reading the flags failed, return error indication now. */
  if (oldflags < 0)
    return oldflags;
  /* Set just the flag we want to set. */
  if (value != 0)
    oldflags |= FD_CLOEXEC;
  else
    oldflags &= ~FD_CLOEXEC;
  /* Store modified flag word in the descriptor. */
  return fcntl (desc, F_SETFD, oldflags);
}

```

I think it passes on arg to the syscall but it is up to the implementation not to do anything with it. like @sbc100  said.

Reply:
Sure there is nothing stopping you passing extra args if you like.. just like calling printf with extra args is harmless.  They problem only happens when printf (or fcntl) reads too far (or you don't provide enough args)... but those cases lead to undefined behaviour IIUC

Reply:
I see, thanks. I was reading the code here, which does read `arg`, but I guess it's ignored later on as mentioned. Sgtm to go with the manpage.